src/translators/rust/pure/cargo-lock/default.nix

```diff
@@ -180,31 +201,24 @@
             path = dependencyObject:
               let
                 findCratePath = name:
-                  l.dirOf (
+                  l.baseNameOf (l.dirOf (
                     l.findFirst
                     (toml: toml.value.package.name == name)
                     (throw "could not find crate ${name}")
                     cargoPackages
-                  ).path;
+                  ).path);
```

Comment:
This will be inefficient if we have many such path-like dependencies as it re-iterates over all possible packages. At some point we probably should generate an attrset containing all the packages to allow for cheap lookups here.

Reply:
Yes, that would be nice. Ideally I would want to only do a one time pass for all the packages, and extract all info we need at once. But for now I just wanted to get it right :D